carbon-design-system/carbon-addons-iot-react
# feat(table): preserve columns widths on resize

Closes #2658

**Summary**
Sibling columns widths are now preserved when a column is resized if option `preserveColumnWidths` is true.

**Change List (commits, features, bugs, etc)**
- Added `preserveColumnWidths` to ColumnResize component to allow sibling columns to maintain their width when a column is resized.
- Added unit tests and renamed some existing to better show the difference between new and old tests


**Acceptance Test (how to verify the PR)**
Go to https://deploy-preview-2666--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-table--with-options-to-explore-column-settings use the default knob configuration

1. resize columns and make sure only the resized column changes in width
   * A column can be made wider, all the way to the end of the right side of the visible part of the table
   * A column can be made thinner, but not thinner than the column min width of 62px.
   * Make sure RTL works
2. combine resizing with adding/removing/toggling columns to make sure it all works together
3. uncheck the preserveColumnWidths and verify that the old resize behaviour is still working


**Regression Test (how to make sure this PR doesn't break old functionality)**

1. Test resizing in other table stories and make sure they work as before



**Things to look for during review**
